common/src/main/java/tachyon/client/FileSystemMasterClient.java

```diff
@@ -79,20 +77,21 @@ protected void afterConnect() {
    * @throws InvalidPathException if the given path is invalid
    * @throws IOException if an I/O error occurs
    */
-  public synchronized long getFileId(String path) throws IOException, InvalidPathException {
-    int retry = 0;
-    while (!mClosed && (retry ++) <= RPC_MAX_NUM_RETRY) {
-      connect();
-      try {
-        return mClient.getFileId(path);
-      } catch (InvalidPathException e) {
-        throw e;
-      } catch (TException e) {
-        LOG.error(e.getMessage(), e);
-        mConnected = false;
-      }
+  public synchronized long getFileId(final String path) throws IOException, InvalidPathException {
+    try {
+      return retryRPC(new RpcCallableWithPropagateTException<Long>() {
+        @Override
+        public Long call() throws PropagateTException, TException {
+          try {
+            return mClient.getFileId(path);
+          } catch (InvalidPathException e) {
+            throw new PropagateTException(e);
+          }
+        }
+      });
+    } catch (RpcCallableWithPropagateTException.PropagateTException e) {
+      throw (InvalidPathException) e.getWrappedTException();
```

Comment:
On a more general note, I suggest you separate your RPC retry logic into a separate PR so that we can merge the raw table changes independent of refining the RPC retry logic.